omp.cache file has old cache entries which cause the prompt to intermittently malfunction

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

Hi @JanDeDobbeleer, I've reproduced this problem on both oh-my-posh v23.14.1 and v13.1.2, so this problem has been around for a long time.

My cmd process id is 42508.
Running `oh-my-posh print primary` should use the default theme.
But instead it uses a seemingly random theme.
But only in one cmd.exe process -- in others it works as expected.

I believe I've tracked it down to a problem in how `omp.cache` is managed:

My `omp.cache` file has 153 entries.
There's an entry for 42508, but it's from May 2022.

```
    "template_cache_42508": {
        "value": "{\"Root\":false,\"PWD\":\"C:\\\\wbin\\\\clink\",\"Folder\":\"clink\",\"Shell\":\"WindowsTerminal\",\"ShellVersion\":\"\",\"UserName\":\"chrisant\",\"HostName\":\"CHRISANT16\",\"Code\":0,\"Env\":{\"4DOS\":\"c:\\\\4dos\",\"ALLUSERSPROFILE\":\"C:\\\\ProgramData\",\"ANSICON\":\"1\",\"APPDATA\":\"C:\\\\Users\\\\chrisant\\\\AppData\\\\Roaming\",\"BAT\":\"c:\\\\4dos\",\"BBDIFF\":\"c:\\\\wbin\\\\x64\\\\sdvdiff.exe --sd\",\"BINNT\":\"c:\\\\binnt\",\"CLINK_BINDIR\":\"c:\\\\wbin\\\\clink\",\"CLINK_COMPLETIONS_DIR\":\"c:\\\\wbin\\\\clink\\\\completions\",\"CLINK_INPUTRC\":\"c:\\\\wbin\\\\clink\",\"CLINK_MYSTYLE\":\"bubbles\",\"COLUMNS\":\"120\",\"COMPUTERNAME\":\"CHRISANT16\",\"ChocolateyInstall\":\"C:\\\\ProgramData\\\\chocolatey\",\"ChocolateyLastPathUpdate\":\"133670655767495766\",\"ComSpec\":\"C:\\\\Windows\\\\system32\\\\cmd.exe\",\"CommonProgramFiles\":\"C:\\\\Program Files\\\\Common Files\",\"CommonProgramFiles(x86)\":\"C:\\\\Program Files (x86)\\\\Common Files\",\"CommonProgramW6432\":\"C:\\\\Program Files\\\\Common Files\",\"DIRX_NERD_FONTS_VERSION\":\"2\",\"DriverData\":\"C:\\\\Windows\\\\System32\\\\Drivers\\\\DriverData\",\"EXT\":\"c:\",\"FZF_ALT_C_COMMAND\":\"c:\\\\wbin\\\\dirx.exe /b /s /X:d /a:d-s-h --bare-relative --utf8 -- $dir\",\"FZF_COMPLETION_OPTS\":\"--preview-window \\\"right:40%,border-left\\\" --bind \\\"ctrl-/:change-preview-window(right:70%|hidden|)\\\" --bind \\\"shift-down:preview-down+preview-down\\\" --bind \\\"shift-up:preview-up+preview-up\\\" --bind \\\"preview-scroll-up:preview-up+preview-up\\\" --bind \\\"preview-scroll-down:preview-down+preview-down\\\" --preview \\\"fzf-preview.cmd {}\\\"\",\"FZF_CTRL_T_COMMAND\":\"c:\\\\wbin\\\\dirx.exe /b /s /X:d /a:-d-s-h --bare-relative --utf8 -- $dir\",\"FZF_CTRL_T_OPTS\":\"--preview-window \\\"right:40%,border-left\\\" --bind \\\"ctrl-/:change-preview-window(right:70%|hidden|)\\\" --bind \\\"shift-down:preview-down+preview-down\\\" --bind \\\"shift-up:preview-up+preview-up\\\" --bind \\\"preview-scroll-up:preview-up+preview-up\\\" --bind \\\"preview-scroll-down:preview-down+preview-down\\\" --preview \\\"fzf-preview.cmd {}\\\"\",\"FZF_DEFAULT_OPTS\":\"--color \\\"border:236,hl:164,hl+:164,fg+:232,bg+:252,gutter:236,info:33,pointer:9\\\"\",\"FZF_ICON_WIDTH\":\"\",\"HOMEDRIVE\":\"C:\",\"HOMEPATH\":\"\\\\Users\\\\chrisant\",\"JJPACK_DIFF\":\"c:\\\\wbin\\\\x64\\\\sdvdiff.exe\",\"LINES\":\"50\",\"LOCALAPPDATA\":\"C:\\\\Users\\\\chrisant\\\\AppData\\\\Local\",\"LOGONSERVER\":\"\\\\\\\\CHRISANT16\",\"MORE\":\"/E\",\"NUMBER_OF_PROCESSORS\":\"22\",\"OS\":\"Windows_NT\",\"OneDrive\":\"C:\\\\Users\\\\chrisant\\\\OneDrive\",\"OneDriveConsumer\":\"C:\\\\Users\\\\chrisant\\\\OneDrive\",\"PATHEXT\":\".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY;.PYW;.LNK\",\"POSH_THEME\":\"c:\\\\wbin\\\\oh-my-posh\\\\lambdageneration.omp.json\",\"PROCESSOR_ARCHITECTURE\":\"AMD64\",\"PROCESSOR_IDENTIFIER\":\"Intel64 Family 6 Model 170 Stepping 4, GenuineIntel\",\"PROCESSOR_LEVEL\":\"6\",\"PROCESSOR_REVISION\":\"aa04\",\"PROMPT\":\"C\\u0008L\\u0008I\\u0008N\\u0008K\\u0008 \\u0008$+$p$g\",\"PSModulePath\":\"C:\\\\Program Files\\\\WindowsPowerShell\\\\Modules;C:\\\\Windows\\\\system32\\\\WindowsPowerShell\\\\v1.0\\\\Modules\",\"PUBLIC\":\"C:\\\\Users\\\\Public\",\"Path\":\"C:\\\\Python312\\\\Scripts\\\\;C:\\\\Python312\\\\;C:\\\\Windows\\\\system32;C:\\\\Windows;C:\\\\Windows\\\\System32\\\\Wbem;C:\\\\Windows\\\\System32\\\\WindowsPowerShell\\\\v1.0\\\\;C:\\\\Windows\\\\System32\\\\OpenSSH\\\\;C:\\\\Program Files\\\\dotnet\\\\;C:\\\\Program Files (x86)\\\\NVIDIA Corporation\\\\PhysX\\\\Common;C:\\\\Program Files\\\\NVIDIA Corporation\\\\NVIDIA NvDLISR;C:\\\\Program Files\\\\Git\\\\cmd;C:\\\\Program Files\\\\Microsoft SQL Server\\\\150\\\\Tools\\\\Binn\\\\;C:\\\\Program Files (x86)\\\\Windows Kits\\\\10\\\\Windows Performance Toolkit\\\\;C:\\\\Program Files\\\\nodejs\\\\;C:\\\\ProgramData\\\\chocolatey\\\\bin;C:\\\\Strawberry\\\\c\\\\bin;C:\\\\Strawberry\\\\perl\\\\site\\\\bin;C:\\\\Strawberry\\\\perl\\\\bin;C:\\\\Program Files\\\\GitHub CLI\\\\;c:\\\\sd\\\\default;c:\\\\4dos;c:\\\\chrisant\\\\vc;c:\\\\beta;c:\\\\wbin;c:\\\\wbin\\\\clink;c:\\\\binnt;c:\\\\sd\\\\SDPack;c:\\\\sd\\\\SDPack\\\\bin;C:\\\\Users\\\\chrisant\\\\AppData\\\\Local\\\\Microsoft\\\\WindowsApps;C:\\\\Users\\\\chrisant\\\\AppData\\\\Local\\\\Programs\\\\Microsoft VS Code\\\\bin;C:\\\\Users\\\\chrisant\\\\.dotnet\\\\tools;C:\\\\Users\\\\chrisant\\\\AppData\\\\Local\\\\Microsoft\\\\WinGet\\\\Packages\\\\Schniz.fnm_Microsoft.Winget.Source_8wekyb3d8bbwe;C:\\\\Users\\\\chrisant\\\\AppData\\\\Roaming\\\\npm;C:\\\\Users\\\\chrisant\\\\AppData\\\\Local\\\\Microsoft\\\\WinGet\\\\Packages\\\\hpjansson.Chafa_Microsoft.Winget.Source_8wekyb3d8bbwe\\\\chafa-1.14.2-1-x86_64-win;\",\"ProgramData\":\"C:\\\\ProgramData\",\"ProgramFiles\":\"C:\\\\Program Files\",\"ProgramFiles(x86)\":\"C:\\\\Program Files (x86)\",\"ProgramW6432\":\"C:\\\\Program Files\",\"SDALIASES\":\"c:\\\\sd\\\\default\\\\sdaliases.ini\",\"SDBIN\":\"c:\\\\sd\\\\default\",\"SDFDIFF\":\"c:\\\\wbin\\\\x64\\\\sdvdiff.exe --sd -LO$c\",\"SDFFONT\":\"Fira Code NF,10\",\"SDFNOWRAP\":\"http\",\"SDPACKBIN\":\"c:\\\\sd\\\\SDPack\",\"SDPOPTIONS\":\"diff -du\",\"SDPWDIFF\":\"c:\\\\wbin\\\\x64\\\\sdvdiff.exe\",\"SDVCDIFF\":\"c:\\\\wbin\\\\x64\\\\sdvdiff.exe --sd -LD\",\"SDVDIFF\":\"c:\\\\wbin\\\\x64\\\\sdvdiff.exe --sd\",\"SESSIONNAME\":\"Console\",\"SystemDrive\":\"C:\",\"SystemRoot\":\"C:\\\\Windows\",\"TEMP\":\"c:\\\\tmp\\\\Temp\",\"TMP\":\"c:\\\\tmp\\\\Temp\",\"TSE\":\"c:\\\\beta\",\"USERDOMAIN\":\"CHRISANT16\",\"USERDOMAIN_ROAMINGPROFILE\":\"CHRISANT16\",\"USERNAME\":\"chrisant\",\"USERPROFILE\":\"C:\\\\Users\\\\chrisant\",\"WBIN\":\"c:\\\\wbin\",\"WSLENV\":\"WT_SESSION:WT_PROFILE_ID:\",\"WT_PROFILE_ID\":\"{55e9b923-7379-4015-8a31-7c40760a0635}\",\"WT_SESSION\":\"b2116767-3295-468b-8aa4-4fc70b7d9918\",\"ZES_ENABLE_SYSMAN\":\"1\",\"_ZL_ADD_ONCE\":\"1\",\"_ZL_FZF\":\"c:\\\\wbin\\\\fzf.exe\",\"_ZL_LUA_EXE\":\"c:\\\\wbin\\\\luabin\\\\lua52.exe\",\"_ZL_MAXAGE\":\"7500\",\"clink_dummy_capture_env\":\" \",\"windir\":\"C:\\\\Windows\"},\"OS\":\"windows\",\"WSL\":false,\"Segments\":{\"Battery\":{\"Percentage\":100,\"State\":2,\"Error\":\"\",\"Icon\":\"ï‰€ \"},\"Executiontime\":{\"FormattedMs\":\"0ms\",\"Ms\":0},\"Exit\":{\"Meaning\":\"0\"},\"Os\":{\"Icon\":\"î˜ª\"},\"Path\":{\"Path\":\"C:\\\\..\\\\clink\",\"StackCount\":0,\"Location\":\"C:\\\\wbin\\\\clink\",\"Writable\":true,\"RootDir\":false},\"Session\":{\"SSHSession\":false,\"DefaultUserName\":\"\"},\"Time\":{\"CurrentDate\":\"2024-09-22T18:09:17.5193368-07:00\",\"Format\":\"15:04:05, _2\"}}}",
        "timestamp": 1727053757,
        "ttl": 1440
    },
```

The timestamp is from May 2022:
```
1653779745:  2022/05/28 23:15:45  [in UTC]
1653754545:  2022/05/28 16:15:45  [converted to Local]
```

The cached environment in the 42508 entry has `\"POSH_THEME\":\"c:\\\\wbin\\\\oh-my-posh\\\\lambdageneration.omp.json\"`, and the omp program interprets that as meaning it should that theme for all cmd.exe processes that ever get assigned a process id 42508 by the OS.

At least on Windows, process id numbers are only unique at a given moment in time.  42508 could be the id of a cmd.exe, but once the cmd.exe process ends then the number can get assigned to any other new process (which may or may not be another cmd.exe process).

I don't know enough about how the `omp.cache` is used, so I don't know specifically what to suggest for how to solve the issue.  I can help brainstorm solutions, if I know more about how it's used.

I assume the omp program has some upper bound for the number of cache entries it will keep, and presumably it ages out old entries.
- Maybe it could age out old entries more aggressively, so it doesn't reuse 2.5 year old entries?  But that won't prevent the problem from also occurring within the shorter age-out timeframe.
- Maybe it could update the cached environment each time, so it doesn't keep using an old environment?

### Theme

That's the problem -- it's picking a theme by mistake from an old cache entry from May 2022.  🙃

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

cmd

### Log output

```Shell
Version: 23.14.1

Shell: cmd

Prompt:

╭─   0ms  C:\wbin\clink   100                                            chrisant@CHRISANT16  18:59:56, 22  
╰─ﬦ

Segments:

ConsoleTitle(false)                        -   0 ms
Os(true)                                   -   2 ms
Executiontime(true)                        -   2 ms
Path(true)                                 -   4 ms
Battery(true)                              -  17 ms
Exit(true)                                 -   2 ms
Session(true)                              -   0 ms
Time(true)                                 -   0 ms
Path(true)                                 -   0 ms

Run duration: 50.1638ms

Cache path: C:\Users\chrisant\AppData\Local\oh-my-posh

Config path: c:\wbin\oh-my-posh\lambdageneration.omp.json

Logs:

[DEBUG] 18:59:56.847 debug.go:49 → debug mode enabled
[DEBUG] 18:59:56.848 debug.go:49 → plain mode enabled
[DEBUG] 18:59:56.848 terminal.go:Getenv:164 → C:\Users\chrisant\AppData\Local
[TRACE] 18:59:56.849 terminal.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 18:59:56.849 terminal.go:CachePath() - 518.1µs
[DEBUG] 18:59:56.849 terminal.go:70 → loading cache file: C:\Users\chrisant\AppData\Local\oh-my-posh\omp.cache
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_34168
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_2500
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_29860
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_35752
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_32808
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_34488
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_42152
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_24764
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_29320
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_35320
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_43228
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_9288
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_26580
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_30160
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_27344
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_40060
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_9420
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_18360
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_27316
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_30704
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_32976
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_36424
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_37588
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_38296
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_39456
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_13876
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_17068
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_41272
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_23080
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_29940
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_29968
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_37116
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_42504
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_43408
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_20352
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_21116
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_30020
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_38624
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_17152
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_21832
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_28224
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_36252
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_4484
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_5872
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_10100
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_12084
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_27764
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_35816
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_39156
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_14588
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_23044
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_36636
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_37784
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_41404
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_12836
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_36452
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_6920
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_30428
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_43424
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_25264
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_39928
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_6204
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_11664
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_16524
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_24924
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_27044
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_39400
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_43576
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_7964
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_21420
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_22496
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_38612
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_36484
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_38364
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_25860
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_32896
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_36188
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_44020
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_13684
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_14236
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_37640
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_40452
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_35216
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_35912
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_31520
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_36140
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_39064
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_40284
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_960
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_13084
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_15424
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_13348
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_27620
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_38536
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_18900
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_19708
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_42816
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_29052
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_36392
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_29784
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_33828
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_37528
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_42228
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_9052
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_12976
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_25992
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_41972
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_42828
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_23220
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_37516
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_26084
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_35372
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_39440
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_41488
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_41840
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_42176
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_17168
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_24728
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_25484
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_32060
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_3424
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_36432
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_36796
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_5804
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_14592
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_17232
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_24604
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_6448
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_38264
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_41344
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_42508
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_22616
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_27448
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_31188
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_43728
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_28472
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_15568
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_21388
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_29716
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_31468
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_3460
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_7504
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_8868
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_16112
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_26372
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_44016
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_8380
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_37696
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_42264
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_37356
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_43944
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_35152
[DEBUG] 18:59:56.857 terminal.go:70 → loading cache key: template_cache_37348
[TRACE] 18:59:56.857 terminal.go(C:\Users\chrisant\AppData\Local\oh-my-posh\omp.cache) - 8.4534ms
[DEBUG] 18:59:56.857 terminal.go:Getenv:164 → C:\Users\chrisant\AppData\Local
[TRACE] 18:59:56.857 terminal.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 18:59:56.858 terminal.go:CachePath() - 1.0497ms
[DEBUG] 18:59:56.858 terminal.go:70 → loading cache file: C:\Users\chrisant\AppData\Local\oh-my-posh\omp.cache.42508
[ERROR] 18:59:56.859 terminal.go:70 → open C:\Users\chrisant\AppData\Local\oh-my-posh\omp.cache.42508: The system cannot find the file specified.
[TRACE] 18:59:56.859 terminal.go(C:\Users\chrisant\AppData\Local\oh-my-posh\omp.cache.42508) - 527.5µs
[TRACE] 18:59:56.859 terminal.go:setPromptCount() - 0s
[DEBUG] 18:59:56.859 terminal.go:Getenv:164 → c:\wbin\oh-my-posh\lambdageneration.omp.json
[TRACE] 18:59:56.859 terminal.go:Getenv(POSH_THEME) - 0s
[DEBUG] 18:59:56.859 terminal.go:ResolveConfigPath:96 → config set using POSH_THEME: c:\wbin\oh-my-posh\lambdageneration.omp.json
[TRACE] 18:59:56.859 terminal.go:ResolveConfigPath() - 0s
[TRACE] 18:59:56.859 terminal.go:Init() - 12.6625ms
[TRACE] 18:59:56.859 terminal.go:Flags() - 0s
[TRACE] 18:59:56.860 load.go:loadConfig() - 520.3µs
[TRACE] 18:59:56.860 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.860 debug.go:57 → terminal program: Windows Terminal
[DEBUG] 18:59:56.860 debug.go:57 → terminal shell: shell
[DEBUG] 18:59:56.860 terminal.go:Getenv:164 → NO DATA
[TRACE] 18:59:56.860 terminal.go:Getenv(OMP_CACHE_DISABLED) - 0s
[TRACE] 18:59:56.860 terminal_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.860 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.860 terminal_windows.go:WindowsRegistryKeyValue:215 → ColorizationColor(DWORD): 0xC4534588
[TRACE] 18:59:56.860 terminal.go:Shell() - 0s
[DEBUG] 18:59:56.860 terminal.go:Getenv:164 → NO DATA
[TRACE] 18:59:56.860 terminal.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 18:59:56.860 debug.go:PrintDebug:23 → segment: Title
[DEBUG] 18:59:56.860 text.go:Render:80 → rendering template:
[TRACE] 18:59:56.860 terminal.go:Flags() - 0s
[TRACE] 18:59:56.860 terminal.go:Shell() - 0s
[DEBUG] 18:59:56.860 terminal.go:Getenv:164 → NO DATA
[TRACE] 18:59:56.860 terminal.go:Getenv(POSH_CURSOR_LINE) - 0s
[DEBUG] 18:59:56.860 terminal.go:Getenv:164 → NO DATA
[TRACE] 18:59:56.860 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 0s
[TRACE] 18:59:56.860 terminal.go:Flags() - 0s
[TRACE] 18:59:56.860 terminal.go:Flags() - 0s
[TRACE] 18:59:56.861 terminal.go:Flags() - 0s
[TRACE] 18:59:56.860 terminal.go:Flags() - 0s
[TRACE] 18:59:56.860 terminal.go:Flags() - 0s
[TRACE] 18:59:56.861 terminal.go:Flags() - 0s
[TRACE] 18:59:56.861 terminal.go:Flags() - 0s
[TRACE] 18:59:56.861 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.861 terminal.go:Flags() - 0s
[TRACE] 18:59:56.861 terminal.go:PathSeparator() - 0s
[TRACE] 18:59:56.861 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.861 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.861 terminal.go:Pwd:189 → C:\wbin\clink
[TRACE] 18:59:56.861 terminal.go:Pwd() - 533.6µs
[DEBUG] 18:59:56.861 terminal_windows.go:58 → C:\Users\chrisant
[TRACE] 18:59:56.861 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.861 segment.go:SetEnabled:115 → segment: Executiontime
[TRACE] 18:59:56.861 terminal.go:Pwd() - 0s
[DEBUG] 18:59:56.861 terminal_windows.go:58 → C:\Users\chrisant
[TRACE] 18:59:56.861 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.861 segment.go:SetEnabled:115 → segment: Battery
[DEBUG] 18:59:56.861 status.go:Init:53 → {{ .Code }}
[TRACE] 18:59:56.861 terminal_windows.go:IsWsl() - 0s
[TRACE] 18:59:56.861 terminal.go:Pwd() - 0s
[DEBUG] 18:59:56.861 terminal_windows.go:58 → C:\Users\chrisant
[TRACE] 18:59:56.861 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.861 segment.go:SetEnabled:115 → segment: Exit
[TRACE] 18:59:56.861 terminal.go:StatusCodes() - 0s
[DEBUG] 18:59:56.861 text.go:Render:80 → rendering template: {{ .Code }}
[TRACE] 18:59:56.861 terminal.go:Flags() - 0s
[TRACE] 18:59:56.861 terminal.go:Pwd() - 0s
[DEBUG] 18:59:56.861 terminal_windows.go:58 → C:\Users\chrisant
[TRACE] 18:59:56.861 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.861 segment.go:SetEnabled:115 → segment: Root
[TRACE] 18:59:56.861 terminal.go:Pwd() - 0s
[DEBUG] 18:59:56.861 terminal_windows.go:58 → C:\Users\chrisant
[TRACE] 18:59:56.861 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.861 segment.go:SetEnabled:115 → segment: Path
[TRACE] 18:59:56.861 terminal.go:Pwd() - 0s
[TRACE] 18:59:56.861 terminal_windows.go:Root() - 0s
[DEBUG] 18:59:56.861 executiontime.go:Enabled:57 → always_enabled: true
[DEBUG] 18:59:56.861 path.go:149 → display_cygpath: false
[TRACE] 18:59:56.861 terminal.go:ExecutionTime() - 0s
[TRACE] 18:59:56.861 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.861 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.861 terminal.go:PathSeparator() - 0s
[TRACE] 18:59:56.861 terminal.go:Pwd() - 0s
[TRACE] 18:59:56.861 terminal.go:Shell() - 0s
[DEBUG] 18:59:56.861 executiontime.go:Enabled:59 → threshold: 500.000000
[TRACE] 18:59:56.861 terminal.go:Shell() - 0s
[DEBUG] 18:59:56.861 executiontime.go:Enabled:63 → austin
[DEBUG] 18:59:56.861 terminal_windows.go:58 → C:\Users\chrisant
[TRACE] 18:59:56.861 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.861 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.861 segment.go:SetEnabled:115 → segment: Os
[TRACE] 18:59:56.861 terminal_windows.go:Root() - 0s
[TRACE] 18:59:56.862 terminal.go:Shell() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:StatusCodes() - 0s
[TRACE] 18:59:56.862 terminal_windows.go:IsWsl() - 0s
[DEBUG] 18:59:56.862 os.go:Enabled:34 → 
[TRACE] 18:59:56.862 terminal.go:Pwd() - 0s
[DEBUG] 18:59:56.862 terminal_windows.go:58 → C:\Users\chrisant
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.862 segment.go:SetEnabled:115 → segment: Git
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.862 terminal.go:Getenv:164 → C:\Users\chrisant\AppData\Local
[TRACE] 18:59:56.862 terminal.go:Getenv(LOCALAPPDATA) - 0s
[DEBUG] 18:59:56.862 terminal.go:TemplateCache:651 → environment: [4DOS=c:\4dos =C:=c:\wbin\clink =clink.bin=c:\wbin\clink =clink.id=42508 =clink.profile=C:\Users\chrisant\AppData\Local\clink =clink.scripts= =cwdhistory_injected=1 =ExitCode=00000000 ALLUSERSPROFILE=C:\ProgramData ANSICON=1 APPDATA=C:\Users\chrisant\AppData\Roaming BAT=c:\4dos BBDIFF=c:\wbin\x64\sdvdiff.exe --sd BINNT=c:\binnt ChocolateyInstall=C:\ProgramData\chocolatey ChocolateyLastPathUpdate=133670655767495766 CLINK_BINDIR=c:\wbin\clink CLINK_COMPLETIONS_DIR=c:\wbin\clink\completions clink_dummy_capture_env=  CLINK_INPUTRC=c:\wbin\clink CLINK_MATCH_COLORS=di=93:ln=94:*.zip=36:ro=32:ro ex=92:ex=1:hi=91:*.arc=35:*.arj=35:*.lzh=35:*.rar=35:*.zoo=35:*.patch=96:*.diff=96:*.dpk=96 CLINK_MYSTYLE=bubbles COLUMNS=120 CommonProgramFiles=C:\Program Files (x86)\Common Files CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files CommonProgramW6432=C:\Program Files\Common Files COMPUTERNAME=CHRISANT16 ComSpec=C:\Windows\system32\cmd.exe DIRX_COLORS=di=1;33:ro=32:ro ex=1;32:ex=1:co=35:hi=38;2;204;64;64:sy=38;2;204;64;64:*.zip=36:sn=38;5;244:nt=38;2;239;65;54:ng=38;2;252;176;64:nm=38;2;240;230;50:nk=38;2;142;198;64:nb=38;2;1;148;68:sb=38;5;240:ut=38;2;159;43;36:ug=38;2;168;117;42:um=38;2;160;153;33:uk=38;2;94;132;42:ub=38;2;0;98;45:da=38;5;33:hM=7:sc=33:bu=33;4:GO=7;38;2;204;204;0:ga=92:gm=38;5;33:gd=38;5;160:gv=93:gt=38;5;164:gi=90:gc=97;48;5;88 DIRX_NERD_FONTS_VERSION=2 DriverData=C:\Windows\System32\Drivers\DriverData EXT=c: FZF_ALT_C_COMMAND=c:\wbin\dirx.exe /b /s /X:d /a:d-s-h --bare-relative --utf8 -- $dir FZF_ALT_C_OPTS=--preview-window "right:40%,border-left" --bind "ctrl-/:change-preview-window(right:70%|hidden|)" --bind "shift-down:preview-down+preview-down" --bind "shift-up:preview-up+preview-up" --bind "preview-scroll-up:preview-up+preview-up" --bind "preview-scroll-down:preview-down+preview-down" --preview "dirx.exe -b -s --bare-relative --utf8 --level=3 --tree --escape-codes=always -- {}" FZF_COMPLETION_OPTS=--preview-window "right:40%,border-left" --bind "ctrl-/:change-preview-window(right:70%|hidden|)" --bind "shift-down:preview-down+preview-down" --bind "shift-up:preview-up+preview-up" --bind "preview-scroll-up:preview-up+preview-up" --bind "preview-scroll-down:preview-down+preview-down" --preview "fzf-preview.cmd {}" FZF_CTRL_T_COMMAND=c:\wbin\dirx.exe /b /s /X:d /a:-d-s-h --bare-relative --utf8 -- $dir FZF_CTRL_T_OPTS=--preview-window "right:40%,border-left" --bind "ctrl-/:change-preview-window(right:70%|hidden|)" --bind "shift-down:preview-down+preview-down" --bind "shift-up:preview-up+preview-up" --bind "preview-scroll-up:preview-up+preview-up" --bind "preview-scroll-down:preview-down+preview-down" --preview "fzf-preview.cmd {}" FZF_DEFAULT_OPTS=--color "border:236,hl:164,hl+:164,fg+:232,bg+:252,gutter:236,info:33,pointer:9" FZF_ICON_WIDTH= HOMEDRIVE=C: HOMEPATH=\Users\chrisant JJPACK_DIFF=c:\wbin\x64\sdvdiff.exe LINES=50 LOCALAPPDATA=C:\Users\chrisant\AppData\Local LOGONSERVER=\\CHRISANT16 LS_COLORS=di=93:ln=94:*.zip=36 MORE=/E NUMBER_OF_PROCESSORS=22 OneDrive=C:\Users\chrisant\OneDrive OneDriveConsumer=C:\Users\chrisant\OneDrive OS=Windows_NT Path=C:\Python312\Scripts\;C:\Python312\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\Git\cmd;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files\nodejs\;C:\ProgramData\chocolatey\bin;C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;C:\Program Files\GitHub CLI\;c:\sd\default;c:\4dos;c:\chrisant\vc;c:\beta;c:\wbin;c:\wbin\clink;c:\binnt;c:\sd\SDPack;c:\sd\SDPack\bin;C:\Users\chrisant\AppData\Local\Microsoft\WindowsApps;C:\Users\chrisant\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\chrisant\.dotnet\tools;C:\Users\chrisant\AppData\Local\Microsoft\WinGet\Packages\Schniz.fnm_Microsoft.Winget.Source_8wekyb3d8bbwe;C:\Users\chrisant\AppData\Roaming\npm;C:\Users\chrisant\AppData\Local\Microsoft\WinGet\Packages\hpjansson.Chafa_Microsoft.Winget.Source_8wekyb3d8bbwe\chafa-1.14.2-1-x86_64-win; PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY;.PYW;.LNK POSH_THEME=c:\wbin\oh-my-posh\lambdageneration.omp.json PROCESSOR_ARCHITECTURE=x86 PROCESSOR_ARCHITEW6432=AMD64 PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 170 Stepping 4, GenuineIntel PROCESSOR_LEVEL=6 PROCESSOR_REVISION=aa04 ProgramData=C:\ProgramData ProgramFiles=C:\Program Files (x86) ProgramFiles(x86)=C:\Program Files (x86) ProgramW6432=C:\Program Files PROMPT=$+$p$g PSModulePath=C:\Program Files\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules PUBLIC=C:\Users\Public SDALIASES=c:\sd\default\sdaliases.ini SDBIN=c:\sd\default SDFDIFF=c:\wbin\x64\sdvdiff.exe --sd -LO$c SDFFONT=Fira Code NF,10 SDFNOWRAP=http SDPACKBIN=c:\sd\SDPack SDPOPTIONS=diff -du SDPWDIFF=c:\wbin\x64\sdvdiff.exe SDVCDIFF=c:\wbin\x64\sdvdiff.exe --sd -LD SDVDIFF=c:\wbin\x64\sdvdiff.exe --sd SESSIONNAME=Console SystemDrive=C: SystemRoot=C:\Windows TEMP=c:\tmp\Temp TMP=c:\tmp\Temp TSE=c:\beta USERDOMAIN=CHRISANT16 USERDOMAIN_ROAMINGPROFILE=CHRISANT16 USERNAME=chrisant USERPROFILE=C:\Users\chrisant WBIN=c:\wbin windir=C:\Windows WSLENV=WT_SESSION:WT_PROFILE_ID: WT_PROFILE_ID={55e9b923-7379-4015-8a31-7c40760a0635} WT_SESSION=b2116767-3295-468b-8aa4-4fc70b7d9918 ZES_ENABLE_SYSMAN=1 _ZL_ADD_ONCE=1 _ZL_FZF=c:\wbin\fzf.exe _ZL_LUA_EXE=c:\wbin\luabin\lua52.exe _ZL_MAXAGE=7500]
[DEBUG] 18:59:56.862 path.go:setMappedLocations:564 → 
[DEBUG] 18:59:56.862 path.go:setMappedLocations:565 → 
[TRACE] 18:59:56.862 terminal.go:Pwd() - 0s
[DEBUG] 18:59:56.862 terminal_windows.go:58 → C:\Users\chrisant
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.862 terminal_windows.go:58 → C:\Users\chrisant
[TRACE] 18:59:56.862 terminal.go:PathSeparator() - 0s
[TRACE] 18:59:56.862 terminal_windows.go:IsWsl() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.862 terminal.go:User:330 → chrisant
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:User() - 0s
[DEBUG] 18:59:56.862 path.go:setMappedLocations:566 → ~
[DEBUG] 18:59:56.862 path.go:setMappedLocations:571 → mapped_locations: map[]
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:PathSeparator() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.862 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.863 terminal.go:PathSeparator() - 523.7µs
[TRACE] 18:59:56.863 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.863 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.863 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.863 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.863 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.863 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.863 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.863 path.go:makeFolderFormatMap:839 → NO DATA
[TRACE] 18:59:56.863 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.863 terminal.go:Host:347 → CHRISANT16
[DEBUG] 18:59:56.863 path.go:setStyle:218 → full
[TRACE] 18:59:56.863 terminal.go:Host() - 578.3µs
[TRACE] 18:59:56.863 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.863 path.go:colorizePath:740 → cycle: []
[DEBUG] 18:59:56.863 terminal.go:Getenv:164 → NO DATA
[DEBUG] 18:59:56.863 path.go:getFolderSeparator:273 → NO DATA
[TRACE] 18:59:56.863 terminal.go:Getenv(SHLVL) - 0s
[DEBUG] 18:59:56.863 path.go:getFolderSeparator:275 → \
[DEBUG] 18:59:56.863 path.go:colorizePath:743 → cycle_folder_separator: false
[DEBUG] 18:59:56.863 path.go:colorizePath:744 → %s
[TRACE] 18:59:56.863 terminal.go:TemplateCache() - 1.6204ms
[DEBUG] 18:59:56.863 path.go:colorizePath:746 → %s
[DEBUG] 18:59:56.863 path.go:colorizePath:747 → %s
[DEBUG] 18:59:56.863 path.go:colorizePath:748 → %s
[TRACE] 18:59:56.863 terminal.go:TemplateCache() - 1.1008ms
[TRACE] 18:59:56.863 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.863 terminal.go:Pwd() - 0s
[TRACE] 18:59:56.863 terminal.go:TemplateCache() - 0s
[TRACE] 18:59:56.863 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.863 terminal.go:StackCount() - 0s
[DEBUG] 18:59:56.863 status.go:Enabled:42 → always_enabled: true
[TRACE] 18:59:56.863 terminal.go:TemplateCache() - 0s
[DEBUG] 18:59:56.865 win32_windows.go:isWriteable:303 → current user is member of S-1-5-32-544
[DEBUG] 18:59:56.865 win32_windows.go:isWriteable:311 ↓
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 18:59:56.866 win32_windows.go:isWriteable:313 → user has write access
[TRACE] 18:59:56.866 terminal_windows.go:DirIsWritable() - 2.5927ms
[TRACE] 18:59:56.866 terminal.go:TemplateCache() - 0s
[TRACE] 18:59:56.878 terminal_windows_nix.go:BatteryState() - 16.2525ms
[DEBUG] 18:59:56.878 battery.go:Enabled:59 → 
[TRACE] 18:59:56.878 terminal.go:TemplateCache() - 0s
[DEBUG] 18:59:56.892 terminal.go:CommandPath:389 → C:\Program Files\Git\cmd\git.exe
[TRACE] 18:59:56.892 terminal.go:CommandPath(git.exe) - 30.0302ms
[TRACE] 18:59:56.892 terminal.go:HasCommand(git.exe) - 30.0302ms
[DEBUG] 18:59:56.892 git.go:shouldDisplay:312 → fetch_bare_info: false
[TRACE] 18:59:56.892 terminal.go:Pwd() - 0s
[DEBUG] 18:59:56.893 terminal.go:ResolveSymlink:286 → C:\wbin\clink
[TRACE] 18:59:56.893 terminal.go:ResolveSymlink(C:\wbin\clink) - 103.8µs
[ERROR] 18:59:56.893 terminal.go:HasParentFilePath:553 → CreateFile .git: The system cannot find the file specified.
[TRACE] 18:59:56.893 terminal.go:HasParentFilePath(.git) - 610µs
[DEBUG] 18:59:56.893 text.go:Render:80 → rendering template:  {{ if gt .Code 0 }} {{ .Meaning }}{{ else }}{{ end }}
[TRACE] 18:59:56.893 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.893 text.go:Render:80 → rendering template:  {{ .FormattedMs }} 
[TRACE] 18:59:56.893 terminal.go:Flags() - 0s
[TRACE] 18:59:56.894 terminal.go:TemplateCache() - 0s
[DEBUG] 18:59:56.894 text.go:Render:80 → rendering template:  {{ if .WSL }}WSL at {{ end }}{{.Icon}} 
[TRACE] 18:59:56.894 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.894 text.go:Render:80 → rendering template:  {{ .Path }}
[DEBUG] 18:59:56.894 text.go:Render:80 → rendering template:  {{ if not .Error }}{{ .Icon }}{{ .Percentage }}{{ end }}{{ .Error }} 
[TRACE] 18:59:56.894 terminal.go:Flags() - 0s
[TRACE] 18:59:56.894 terminal.go:Flags() - 0s
[TRACE] 18:59:56.894 terminal.go:TemplateCache() - 0s
[TRACE] 18:59:56.894 terminal.go:TemplateCache() - 0s
[TRACE] 18:59:56.894 terminal.go:TemplateCache() - 0s
[TRACE] 18:59:56.894 terminal.go:TemplateCache() - 0s
[DEBUG] 18:59:56.894 text.go:Render:80 → rendering template: plain
[TRACE] 18:59:56.894 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.894 text.go:Render:80 → rendering template: plain
[TRACE] 18:59:56.894 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.894 text.go:Render:80 → rendering template: diamond
[TRACE] 18:59:56.894 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.894 text.go:Render:80 → rendering template: powerline
[TRACE] 18:59:56.894 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.894 text.go:Render:80 → rendering template: diamond
[TRACE] 18:59:56.895 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.895 text.go:Render:80 → rendering template: {{ if eq "Full" .State.String }}#292929{{ end }}
[TRACE] 18:59:56.895 terminal.go:Flags() - 0s
[TRACE] 18:59:56.895 terminal.go:TemplateCache() - 0s
[DEBUG] 18:59:56.895 text.go:Render:80 → rendering template: powerline
[TRACE] 18:59:56.895 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.895 text.go:Render:80 → rendering template: {{ if eq "Full" .State.String }}#292929{{ end }}
[TRACE] 18:59:56.895 terminal.go:Flags() - 0s
[TRACE] 18:59:56.895 terminal.go:TemplateCache() - 0s
[DEBUG] 18:59:56.895 text.go:Render:80 → rendering template: {{ if gt .Code 0 }}#292929{{ end }}
[TRACE] 18:59:56.895 terminal.go:Flags() - 0s
[TRACE] 18:59:56.895 terminal.go:TemplateCache() - 0s
[DEBUG] 18:59:56.895 text.go:Render:80 → rendering template: diamond
[TRACE] 18:59:56.895 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.895 text.go:Render:80 → rendering template: {{ if gt .Code 0 }}#292929{{ end }}
[TRACE] 18:59:56.895 terminal.go:Flags() - 0s
[TRACE] 18:59:56.895 terminal.go:TemplateCache() - 0s
[TRACE] 18:59:56.895 terminal.go:Shell() - 0s
[TRACE] 18:59:56.895 terminal.go:Shell() - 0s
[TRACE] 18:59:56.895 terminal.go:Flags() - 0s
[TRACE] 18:59:56.895 terminal.go:Pwd() - 0s
[DEBUG] 18:59:56.895 terminal_windows.go:58 → C:\Users\chrisant
[TRACE] 18:59:56.895 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.895 segment.go:SetEnabled:115 → segment: Session
[DEBUG] 18:59:56.895 terminal.go:Getenv:164 → NO DATA
[TRACE] 18:59:56.895 terminal.go:Getenv(SSH_CONNECTION) - 0s
[DEBUG] 18:59:56.895 terminal.go:Getenv:164 → NO DATA
[TRACE] 18:59:56.895 terminal.go:Getenv(SSH_CLIENT) - 0s
[TRACE] 18:59:56.895 terminal.go:TemplateCache() - 0s
[TRACE] 18:59:56.895 terminal.go:Flags() - 0s
[TRACE] 18:59:56.895 terminal.go:Pwd() - 0s
[DEBUG] 18:59:56.895 terminal_windows.go:58 → C:\Users\chrisant
[TRACE] 18:59:56.895 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.895 segment.go:SetEnabled:115 → segment: Time
[DEBUG] 18:59:56.895 time.go:Enabled:29 → 15:04:05, _2
[TRACE] 18:59:56.895 terminal.go:TemplateCache() - 0s
[DEBUG] 18:59:56.895 text.go:Render:80 → rendering template: {{ .CurrentDate | date .Format }} 
[TRACE] 18:59:56.895 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.895 text.go:Render:80 → rendering template:  {{ if .SSHSession }} {{ end }}{{ .UserName }}@{{ .HostName }} <#fb7e14></>
[TRACE] 18:59:56.895 terminal.go:Flags() - 0s
[TRACE] 18:59:56.895 terminal.go:TemplateCache() - 0s
[TRACE] 18:59:56.896 terminal.go:TemplateCache() - 0s
[DEBUG] 18:59:56.896 text.go:Render:80 → rendering template: diamond
[TRACE] 18:59:56.896 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.896 text.go:Render:80 → rendering template: diamond
[TRACE] 18:59:56.896 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.896 terminal_windows.go:TerminalWidth:116 → terminal width: 120
[TRACE] 18:59:56.896 terminal_windows.go:TerminalWidth() - 0s
[TRACE] 18:59:56.896 terminal.go:Shell() - 0s
[TRACE] 18:59:56.896 terminal.go:Shell() - 0s
[TRACE] 18:59:56.896 terminal.go:Flags() - 0s
[TRACE] 18:59:56.896 terminal.go:Pwd() - 0s
[DEBUG] 18:59:56.896 terminal_windows.go:58 → C:\Users\chrisant
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.896 segment.go:SetEnabled:115 → segment: Path
[DEBUG] 18:59:56.896 path.go:149 → display_cygpath: false
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:PathSeparator() - 0s
[TRACE] 18:59:56.896 terminal.go:Pwd() - 0s
[TRACE] 18:59:56.896 terminal.go:Shell() - 0s
[TRACE] 18:59:56.896 terminal.go:Shell() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.896 path.go:setMappedLocations:563 → mapped_locations_enabled: true
[DEBUG] 18:59:56.896 path.go:setMappedLocations:564 → 
[DEBUG] 18:59:56.896 path.go:setMappedLocations:565 → 
[DEBUG] 18:59:56.896 terminal_windows.go:58 → C:\Users\chrisant
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:PathSeparator() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.896 path.go:setMappedLocations:566 → ~
[DEBUG] 18:59:56.896 path.go:setMappedLocations:571 → mapped_locations: map[]
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:PathSeparator() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:PathSeparator() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.896 path.go:makeFolderFormatMap:839 → NO DATA
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[DEBUG] 18:59:56.896 path.go:setStyle:218 → agnoster
[DEBUG] 18:59:56.896 path.go:getAgnosterPath:330 → ..
[DEBUG] 18:59:56.896 path.go:colorizePath:740 → cycle: []
[DEBUG] 18:59:56.896 path.go:getFolderSeparator:273 → NO DATA
[DEBUG] 18:59:56.896 path.go:getFolderSeparator:275 → \
[DEBUG] 18:59:56.896 path.go:colorizePath:743 → cycle_folder_separator: false
[DEBUG] 18:59:56.896 path.go:colorizePath:744 → %s
[DEBUG] 18:59:56.896 path.go:colorizePath:746 → %s
[DEBUG] 18:59:56.896 path.go:colorizePath:747 → %s
[DEBUG] 18:59:56.896 path.go:colorizePath:748 → %s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:Pwd() - 0s
[TRACE] 18:59:56.896 terminal.go:TemplateCache() - 0s
[TRACE] 18:59:56.896 terminal.go:GOOS() - 0s
[TRACE] 18:59:56.896 terminal.go:StackCount() - 0s
[DEBUG] 18:59:56.896 win32_windows.go:isWriteable:303 → current user is member of S-1-5-32-544
[DEBUG] 18:59:56.896 win32_windows.go:isWriteable:311 ↓
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 18:59:56.896 win32_windows.go:isWriteable:313 → user has write access
[TRACE] 18:59:56.896 terminal_windows.go:DirIsWritable() - 0s
[TRACE] 18:59:56.896 terminal.go:TemplateCache() - 0s
[DEBUG] 18:59:56.896 text.go:Render:80 → rendering template: ﬦ
[TRACE] 18:59:56.896 terminal.go:Flags() - 0s
[DEBUG] 18:59:56.896 text.go:Render:80 → rendering template: diamond
[TRACE] 18:59:56.896 terminal.go:Flags() - 0s
[TRACE] 18:59:56.896 terminal.go:Shell() - 0s
[TRACE] 18:59:56.896 terminal.go:Shell() - 0s
[TRACE] 18:59:56.896 terminal.go:Shell() - 0s
[DEBUG] 18:59:56.897 terminal.go:Getenv:164 → C:\Users\chrisant\AppData\Local
[TRACE] 18:59:56.897 terminal.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 18:59:56.897 terminal.go:CachePath() - 519.8µs
[TRACE] 18:59:56.897 terminal.go:Flags() - 0s
```

--- a/src/cache/cache.go
+++ b/src/cache/cache.go
@@ -30,9 +30,9 @@ const (
 var SessionFileName = fmt.Sprintf("%s.%s", FileName, sessionID())
 
 func sessionID() string {
-	pid := os.Getenv("POSH_PID")
+	pid := os.Getenv("POSH_SESSION_ID")
 	if len(pid) == 0 {
-		log.Debug("POSH_PID not set, using process pid")
+		log.Debug("POSH_SESSION_ID not set, using PID")
 		pid = strconv.Itoa(os.Getppid())
 	}
 

--- a/src/go.mod
+++ b/src/go.mod
@@ -10,7 +10,7 @@ require (
 	github.com/alecthomas/repr v0.4.0 // indirect
 	github.com/esimov/stackblur-go v1.1.0
 	github.com/fogleman/gg v1.3.0
-	github.com/google/uuid v1.6.0 // indirect
+	github.com/google/uuid v1.6.0
 	github.com/gookit/color v1.5.4
 	github.com/huandu/xstrings v1.5.0 // indirect
 	github.com/mitchellh/copystructure v1.2.0 // indirect

--- a/src/shell/init.go
+++ b/src/shell/init.go
@@ -6,6 +6,7 @@ import (
 	"strings"
 	"time"
 
+	"github.com/google/uuid"
 	"github.com/jandedobbeleer/oh-my-posh/src/log"
 	"github.com/jandedobbeleer/oh-my-posh/src/runtime"
 )
@@ -80,6 +81,7 @@ func PrintInit(env runtime.Environment, features Features, startTime *time.Time)
 
 	shell := env.Flags().Shell
 	configFile := env.Flags().Config
+	sessionID := uuid.NewString()
 
 	var script string
 
@@ -87,38 +89,47 @@ func PrintInit(env runtime.Environment, features Features, startTime *time.Time)
 	case PWSH, PWSH5:
 		executable = quotePwshOrElvishStr(executable)
 		configFile = quotePwshOrElvishStr(configFile)
+		sessionID = quotePwshOrElvishStr(sessionID)
 		script = pwshInit
 	case ZSH:
 		executable = QuotePosixStr(executable)
 		configFile = QuotePosixStr(configFile)
+		sessionID = QuotePosixStr(sessionID)
 		script = zshInit
 	case BASH:
 		executable = QuotePosixStr(executable)
 		configFile = QuotePosixStr(configFile)
+		sessionID = QuotePosixStr(sessionID)
 		script = bashInit
 	case FISH:
 		executable = quoteFishStr(executable)
 		configFile = quoteFishStr(configFile)
+		sessionID = quoteFishStr(sessionID)
 		script = fishInit
 	case CMD:
 		executable = escapeLuaStr(executable)
 		configFile = escapeLuaStr(configFile)
+		sessionID = escapeLuaStr(sessionID)
 		script = cmdInit
 	case NU:
 		executable = quoteNuStr(executable)
 		configFile = quoteNuStr(configFile)
+		sessionID = quoteNuStr(sessionID)
 		script = nuInit
 	case TCSH:
 		executable = quoteCshStr(executable)
 		configFile = quoteCshStr(configFile)
+		sessionID = quoteCshStr(sessionID)
 		script = tcshInit
 	case ELVISH:
 		executable = quotePwshOrElvishStr(executable)
 		configFile = quotePwshOrElvishStr(configFile)
+		sessionID = quotePwshOrElvishStr(sessionID)
 		script = elvishInit
 	case XONSH:
 		executable = quotePythonStr(executable)
 		configFile = quotePythonStr(configFile)
+		sessionID = quotePythonStr(sessionID)
 		script = xonshInit
 	default:
 		return fmt.Sprintf("echo \"No initialization script available for %s\"", shell)
@@ -128,6 +139,7 @@ func PrintInit(env runtime.Environment, features Features, startTime *time.Time)
 		"::OMP::", executable,
 		"::CONFIG::", configFile,
 		"::SHELL::", shell,
+		"::SESSION_ID::", sessionID,
 	).Replace(script)
 
 	shellScript := features.Lines(shell).String(init)

--- a/src/shell/scripts/omp.bash
+++ b/src/shell/scripts/omp.bash
@@ -1,13 +1,9 @@
 export POSH_THEME=::CONFIG::
 export POSH_SHELL_VERSION=$BASH_VERSION
 export POWERLINE_COMMAND='oh-my-posh'
-export POSH_PID=$$
+export POSH_SESSION_ID=::SESSION_ID::
 export CONDA_PROMPT_MODIFIER=false
 export OSTYPE=$OSTYPE
-
-if [[ $OSTYPE =~ ^(msys|cygwin) ]]; then
-    export POSH_PID=$(command cat /proc/$$/winpid)
-fi
 
 # global variables
 _omp_start_time=''

--- a/src/shell/scripts/omp.elv
+++ b/src/shell/scripts/omp.elv
@@ -1,4 +1,4 @@
-set-env POSH_PID (to-string (randint 10000000000000 10000000000000000))
+set-env POSH_SESSION_ID ::SESSION_ID::
 set-env POSH_THEME ::CONFIG::
 set-env POSH_SHELL_VERSION $version
 set-env POWERLINE_COMMAND oh-my-posh

--- a/src/shell/scripts/omp.fish
+++ b/src/shell/scripts/omp.fish
@@ -1,16 +1,12 @@
 set --export POSH_THEME ::CONFIG::
 set --export POSH_SHELL_VERSION $FISH_VERSION
 set --export POWERLINE_COMMAND oh-my-posh
-set --export POSH_PID $fish_pid
+set --export POSH_SESSION_ID ::SESSION_ID::
 set --export CONDA_PROMPT_MODIFIER false
+
 set --global _omp_tooltip_command ''
 set --global _omp_current_rprompt ''
 set --global _omp_transient 0
-
-if command uname -s | string match -qr '^(CYGWIN|MSYS|MINGW)'
-    set --export POSH_PID (command cat /proc/$fish_pid/winpid)
-end
-
 set --global _omp_executable ::OMP::
 set --global _omp_ftcs_marks 0
 set --global _omp_transient_prompt 0

--- a/src/shell/scripts/omp.lua
+++ b/src/shell/scripts/omp.lua
@@ -3,7 +3,7 @@
 ---@diagnostic disable: lowercase-global
 
 -- Cache PID
-os.setenv('POSH_PID', os.getpid())
+os.setenv('POSH_SESSION_ID', '::SESSION_ID::')
 
 -- Helper functions
 

--- a/src/shell/scripts/omp.nu
+++ b/src/shell/scripts/omp.nu
@@ -6,7 +6,7 @@ if ($env.config? | is-not-empty) {
 $env.POWERLINE_COMMAND = 'oh-my-posh'
 $env.POSH_THEME = (echo ::CONFIG::)
 $env.PROMPT_INDICATOR = ""
-$env.POSH_PID = (random uuid)
+$env.POSH_SESSION_ID = (echo ::SESSION_ID::)
 $env.POSH_SHELL_VERSION = (version | get version)
 
 let _omp_executable: string = (echo ::OMP::)

--- a/src/shell/scripts/omp.ps1
+++ b/src/shell/scripts/omp.ps1
@@ -40,7 +40,7 @@ New-Module -Name "oh-my-posh-core" -ScriptBlock {
 
     $env:POWERLINE_COMMAND = "oh-my-posh"
     $env:POSH_SHELL_VERSION = $script:PSVersion
-    $env:POSH_PID = $PID
+    $env:POSH_SESSION_ID = ::SESSION_ID::
     $env:CONDA_PROMPT_MODIFIER = $false
 
     # set the default theme

--- a/src/shell/scripts/omp.tcsh
+++ b/src/shell/scripts/omp.tcsh
@@ -1,10 +1,8 @@
 setenv POWERLINE_COMMAND "oh-my-posh";
 setenv POSH_THEME ::CONFIG::;
 setenv POSH_SHELL_VERSION "$tcsh";
-setenv POSH_PID $$;
+setenv POSH_SESSION_ID ::SESSION_ID::;
 setenv OSTYPE "$OSTYPE";
-
-if ( "$OSTYPE" =~ {msys,cygwin}* ) setenv POSH_PID "`cat /proc/$$/winpid`";
 
 if ( ! $?_omp_enabled ) alias precmd '
   set _omp_status = $status;

--- a/src/shell/scripts/omp.xsh
+++ b/src/shell/scripts/omp.xsh
@@ -1,8 +1,6 @@
-import uuid
-
 $POWERLINE_COMMAND = "oh-my-posh"
 $POSH_THEME = ::CONFIG::
-$POSH_PID = uuid.uuid4().hex
+$POSH_SESSION_ID = ::SESSION_ID::
 $POSH_SHELL_VERSION = $XONSH_VERSION
 
 _omp_executable = ::OMP::

--- a/src/shell/scripts/omp.zsh
+++ b/src/shell/scripts/omp.zsh
@@ -1,14 +1,10 @@
 export POSH_THEME=::CONFIG::
 export POSH_SHELL_VERSION=$ZSH_VERSION
-export POSH_PID=$$
+export POSH_SESSION_ID=::SESSION_ID::
 export POWERLINE_COMMAND='oh-my-posh'
 export CONDA_PROMPT_MODIFIER=false
 export ZLE_RPROMPT_INDENT=0
 export OSTYPE=$OSTYPE
-
-if [[ $OSTYPE =~ ^(msys|cygwin) ]]; then
-  export POSH_PID=$(command cat /proc/$$/winpid)
-fi
 
 _omp_executable=::OMP::
 _omp_tooltip_command=''

